src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -203,9 +225,9 @@ internal string GetSchemaFromDatabaseQualifiedName(string fullObjectName, string
             string[] tokens = fullObjectName.Split('.');
             for (int i = tokens.Length - 1; i > 0; i--)
             {
-                if(tokens[i].Equals(objectName))
+                if (tokens[i].Equals(objectName))
                 {
-                    return tokens[i-1];
+                    return tokens[i - 1];
```

Comment:
ok so my problem is that you're getting tokens[i - 1] without checking if the tokens has more than item 

---
In reply to: [92512474](https://github.com/Microsoft/sqltoolsservice/pull/177#discussion_r92512474) [](ancestors = 92512474)